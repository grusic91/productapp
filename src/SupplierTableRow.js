import React from 'react';

export class SupplierTableRow extends React.Component {
    render(){
        let s = this.props.supplier;
        return <tr>
            <td>{s.id}</td>
            <td>{s.name}</td>
            <td>{s.city}</td>
            <td>{s.products.join(", ")}</td>
            <td>
                <button 
                    className="btn btn-sm btn-warning m-1" 
                    onClick={() => this.props.editCallback(s)}
                >
                    Edit
                </button>
                <button 
                    className="btn btn-sm btn-danger m-1" 
                    onClick={() => this.props.deleteCallback(s)}
                >
                    Delete
                </button>
            </td>
        </tr>
    }
}