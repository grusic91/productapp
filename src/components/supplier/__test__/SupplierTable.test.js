import React from 'react';
import { shallow } from 'enzyme';
import { SupplierTable } from '../SupplierTable';
import { SupplierTableRow } from '../SupplierTableRow';

describe('<SupplierTable />', () => {
    let wrapper;
    beforeAll(() => wrapper = shallow(<SupplierTable 
        suppliers={[]}
        editCallback={jest.fn()}
        deleteCallback={jest.fn()}
    />));

    it("renders Components properly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("renders table element ", () => {
        expect(wrapper.find('table').length).toEqual(1);
    });

    it("find th element where text is Suppliers", () => {
        expect(wrapper.find('th').first().text()).toEqual("Suppliers");
    });

    it("renders all matching headers of columns", () => {
        expect(wrapper.containsMatchingElement(
            <tr>
                <th>ID</th><th>Name</th><th>City</th><th>Products</th>
            </tr>
        )).toEqual(true);
    });

    it("renders tbody element", () => {
        expect(wrapper.find('tbody')).toHaveLength(1);
    });

    it("in case there are no suppliers, SupplierTableRow is not rendered", () => {
        expect(wrapper.containsMatchingElement(<SupplierTableRow />)).toEqual(false);
    });

    it("in case we have two suppliers SupplierTable rneders two SupplierTableRow components", () => {
        // populate products props on ProductTable
        wrapper.setProps({
            suppliers: [
                {
                    id: 1,
                    name: "Surf Dudes",
                    city: "San Jose",
                    products: [1, 2]
                  },
                  {
                    id: 2,
                    name: "Field Supplies",
                    city: "New York",
                    products: [3]
                  }
            ]
        });
        expect(wrapper.find(SupplierTableRow)).toHaveLength(2);
    });
})
