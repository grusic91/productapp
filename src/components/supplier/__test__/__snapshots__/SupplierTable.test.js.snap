// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SupplierTable /> renders Components properly 1`] = `ShallowWrapper {}`;
